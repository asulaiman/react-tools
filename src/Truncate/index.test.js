/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import cases from 'jest-in-case'
import { Truncate } from './'

describe('Truncate', () => {
  cases(
    'matches snapshots',
    opts => {
      expect(
        renderer
          .create(
            <Truncate>
              "The quick brown fox jumps over the lazy dog" is an
              English-language pangramâ€”a sentence that contains all of the
              letters of the alphabet. It is commonly used for touch-typing
              practice, testing typewriters and computer keyboards, displaying
              examples of fonts, and other applications involving text where the
              use of all letters in the alphabet is desired. Owing to its
              brevity and coherence, it has become widely known.
            </Truncate>
          )
          .toJSON()
      ).toMatchSnapshot()
    },
    [
      { name: 'basic', props: {} },
      { name: 'inline', props: { inline: true } },
      { name: 'custom truncate point', props: { length: 10 } },
      { name: 'custom fudge factor', props: { fudgeFactor: 1.5 } }
    ]
  )
})
