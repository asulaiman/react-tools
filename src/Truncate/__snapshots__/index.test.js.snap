// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Truncate matches snapshots basic 1`] = `
<div
  className="truncate"
>
  "The quick brown fox jumps over the lazy dog" is an English-language pangram—a sentence that contains
  <div
    className="expanderWrapper "
  >
    <div
      className="expander"
      onClick={[Function]}
    >
      <span
        className="dots"
      >
        ...
      </span>
    </div>
  </div>
  <div />
</div>
`;

exports[`Truncate matches snapshots custom fudge factor 1`] = `
<div
  className="truncate"
>
  "The quick brown fox jumps over the lazy dog" is an English-language pangram—a sentence that contains
  <div
    className="expanderWrapper "
  >
    <div
      className="expander"
      onClick={[Function]}
    >
      <span
        className="dots"
      >
        ...
      </span>
    </div>
  </div>
  <div />
</div>
`;

exports[`Truncate matches snapshots custom truncate point 1`] = `
<div
  className="truncate"
>
  "The quick brown fox jumps over the lazy dog" is an English-language pangram—a sentence that contains
  <div
    className="expanderWrapper "
  >
    <div
      className="expander"
      onClick={[Function]}
    >
      <span
        className="dots"
      >
        ...
      </span>
    </div>
  </div>
  <div />
</div>
`;

exports[`Truncate matches snapshots inline 1`] = `
<div
  className="truncate"
>
  "The quick brown fox jumps over the lazy dog" is an English-language pangram—a sentence that contains
  <div
    className="expanderWrapper "
  >
    <div
      className="expander"
      onClick={[Function]}
    >
      <span
        className="dots"
      >
        ...
      </span>
    </div>
  </div>
  <div />
</div>
`;
